youtube_dl/extractor/gramofononline.py

```diff
@@ -0,0 +1,83 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+import re
+import json
+
+
+class GramofonOnlineIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?gramofononline\.hu(/(listen.php\?.*track=)?(?P<id>[0-9]+))?'
+
+    _TESTS = [{
+        'url': 'https://gramofononline.hu/1401835664/papageno-duett',
+        'md5': '1b4bcabde313f09cdd48c463b54d8125',
+        'info_dict': {
+            'id': '1401835664',
+            'title': 'Papageno-Duett ',
+            'artist': 'Johanna Gadski, Otto Goritz, ismeretlen zenekar',
+            'ext': 'mp3'
+        }
+    }, {
+        'url': 'https://gramofononline.hu/listen.php?autoplay=true&track=1401835664',
+        'md5': '1b4bcabde313f09cdd48c463b54d8125',
+        'info_dict': {
+            'id': '1401835664',
+            'title': 'Papageno-Duett ',
+            'artist': 'Johanna Gadski, Otto Goritz, ismeretlen zenekar',
+            'ext': 'mp3'
+        },
+        'params': {
+            'skip_download': True,
+        }
+    }]
+
+    def _get_entry(self, obj):
+        id1 = obj.get("id")
+        source = obj.get("source")
+        name = obj.get("name")
```

Comment:
Mandatory. Read coding conventions.

Reply:
could you please elaborate on this point?

Reply:
Again: read. It's all explained.